Beta_Integrations/integration-Vertica.yml

```diff
@@ -0,0 +1,178 @@
+        return connection
+
+
+    ''' COMMANDS + QUERY FUNCTIONS '''
+
+    def test_module():
+        """
+        Performs basic query on default system tables
+        """
+        cursor.execute("SELECT * FROM system_tables ORDER BY table_schema, table_name LIMIT 2;")
+        rows = cursor.fetchall()
+        if cursor.rowcount == 0:
+            return_error('No results were returned from the DB.')
+        demisto.results('ok')
+
+
+    def query_command():
+        """
+        Execute a query against the DB
+        """
+        # Init main vars
+        contents = []
+        context = {}
+        title = 'Vertica Query'
+
+        # Get arguments from user
+        query = demisto.args().get('query')
+        limit = int(demisto.args().get('limit', 50))
+        # Query and get raw response (list of ordered dicts)
+        rows = query_request(query)
+
+        # Parse response into context & content entries
+        if rows:
+            if limit:
+                rows = rows[:limit]
+            title = 'Vertica Query Results:'
+
+            for i, row in enumerate(rows):
+                rows[i] = { underscoreToCamelCase(k): convert_datetime_to_string(v) for k, v in row.items() }
+
+            contents = rows
+            context['Vertica.Query(val.Query && val.Query === obj.Query)'] = {
+                'Query': query,
+                'Rows': rows
+            }
+
+        demisto.results({
+            'Type': entryTypes['note'],
+            'ContentsFormat': formats['json'],
+            'Contents': contents,
+            'ReadableContentsFormat': formats['markdown'],
+            'HumanReadable': tableToMarkdown(title, contents, removeNull=True),
+            'EntryContext': context
+        })
+
+
+    def query_request(query):
+        try:
+            cursor.execute(query)
+        except vertica_python.errors.MissingRelation:
+            return_error('Error while executing query.')
+        rows = cursor.fetchall()
+        if cursor.rowcount in {0, -1}:
+            return False
+        else:
+            return rows
+
+
+    ''' COMMANDS MANAGER / SWITCH PANEL '''
+
+    LOG('Command being called is %s' % (demisto.command()))
+
+    try:
+        connection = connect_db()
+        cursor = connection.cursor('dict')
+        if demisto.command() == 'test-module':
+            test_module()
+        elif demisto.command() == 'vertica-query':
+            query_command()
+    # Log exceptions
+    except Exception as e:
+        LOG(e)
+        LOG.print_log()
+        raise
+    finally:
+        if 'connection' in globals():
+            connection.close()
+  type: python
+  commands:
+  - name: vertica-query
+    arguments:
+    - name: query
+      required: true
+      description: SQL query to perform on DB
+    - name: limit
+      description: Limit number of results to be returned from the query
+      defaultValue: "50"
+    outputs:
+    - contextPath: Vertica.Query.Query
```

Comment:
not sure i understand why we need two levels here
I think `Vertica.Query` and `Vertica.Row` should do
wdyt ?

Reply:
Mainly to differentiate in case we add more context paths to it in the future, I thought better safe then sorry.

Reply:
But since it's main use is query, we could remove the second level without worrying too much.

Reply:
i think we can remove, its db, so there isn't much more than queries here